### ISSUE TITLE
Linking to the published version of a manuscript

### ISSUE THREAD

**Issue by agitter** [manubot/rootstock#347]
One great feature of bioRxiv is that it automatically adds a link to the published version of a manuscript after publication.  Could we add something similar to Manubot manuscripts?  We already have done this manually for manuscripts like the [meta review](https://greenelab.github.io/meta-review/)
> A DOI-citable version of this manuscript is available in PLOS Computational Biology at https://doi.org/10.1371/journal.pcbi.1007128. This version of the manuscript [contains changes](https://github.com/greenelab/meta-review/compare/b90923a99208c9152cec7929399ab1b7929cb7e2...2482af490804891c7f00b248c4e7ef7f356a0730) subsequent to the journal publication.

The most reliable way to do this would be to add support for a `preprint_citation` and `journal_citation` in the manuscript metadata.  Authors would have to manually add those citations.  A more ambitious and less reliable approach would be to use Crossref or some other API to search the manuscript title and infer a match.

**Comment by dhimmel** [comment]
> The most reliable way to do this would be to add support for a `preprint_citation` and `journal_citation` in the manuscript metadata.

I like this approach. I think it makes sense to think about it with respect to all the other metadata changes we'd like in https://github.com/manubot/manubot/issues/187.

```yaml
citations:
- label: preprint
  citekey: doi:10.xxxx
- label: preprint
  citekey: https://xxxx
- label: journal article
  citekey: doi:10.yyyy
```

This would allow a little more flexibility for arbitrary names like `preprint` / `book` / `newspaper article`. Also would allow multiple preprints.

The front-matter template would insert content like:

***

This manuscript is available as a preprint:

reference goes here

This manuscript is available as a preprint:

reference goes here

This manuscript is available as a journal article:

reference goes here

***

Or some other less verbose format

**Comment by agitter** [comment]
Yes, that's similar to what I had in mind.  Using more flexible labeling makes sense.  Are these "citations" or some more general type of cross-reference?  I haven't looked at what terminology is used elsewhere.

**Comment by dhimmel** [comment]
> Are these "citations" or some more general type of cross-reference?

I was thinking regular manubot citations, possibly with access to manual reference metadata.

**Comment by agitter** [comment]
To expand on that thought, I was wondering if the yaml key should be `citations` or `crossreferences`.  In my mind, a citation would indicate that these are alternative versions of the same manuscript that one may want to cite instead of the Manubot version.

The cross-references could be more general and refer to several related citekeys.  Maybe an author would cross reference the preprint, journal version, PubMed abstract, PubMed Central version, or even related GitHub repositories or dataset DOIs.  That's a broader use case than I originally proposed.  I don't see a specific need for it, but the yaml list of citations would support all of these.
